### ISSUE TITLE
are `abstract` / `base` specifiers permitted or required on class forward declarations?

### ISSUE THREAD

**Issue by zygoloid** [carbon-language/carbon-lang#3384]
### Summary of issue:

Given this class:
```carbon
base class C {}
```
how does one write a forward declaration? Is
```carbon
base class C;
```
valid? Is
```carbon
class C;
```
valid?

### Details:

There doesn't seem to be a reason to require the `abstract` or `base` specifier to be present on a forward declaration. But usually our forward declaration syntax for classes is a prefix of the definition, and there also doesn't seem to be any particular hardship to developers in requiring the forward declaration to repeat the specifier.

In addition to the three answers suggested by the question (specifier is disallowed, specifier is allowed but not required, specifier is required), we could change the class definition syntax to put the specifier in the class body, as we did for inheritance:

```carbon
class C {
  base class;
  extend base: B;
}
```

**Comment by justzh** [comment]
change it

**Comment by chandlerc** [comment]
@justzh - Please keep comments on PRs and issues constructive and on-topic. I'm hiding yours, but you've had warnings now and so I will block if this continues.

**Comment by josh11b** [comment]
This question was discussed at the end of [open discussion on 2024-05-09](https://docs.google.com/document/d/1s3mMCupmuSpWOFJGnvjoElcBIe2aoaysTIdyczvKX84/edit?pli=1&resourcekey=0-G095Wc3sR6pW1hLJbGgE0g&tab=t.0#heading=h.bu7djkos4xo). I believe the conclusions were:
* modifiers on forward declarations should match those on `extern` declarations (except for the `extern` modifier)
* otherwise prefer not repeating modifiers
* prefer putting modifiers on definitions over forward declarations

**Comment by zygoloid** [comment]
This question was [addressed by #3762](https://docs.carbon-lang.dev/proposals/p3762.html#modifier-keywords) after [consideration of alternatives](https://docs.carbon-lang.dev/proposals/p3762.html#other-modifier-keyword-merging-approaches):

> - Other class, impl, and interface modifiers (abstract, base, final) exist only on the definition, not on the forward declaration.
